zebrad/src/components/health.rs

```diff
@@ -0,0 +1,457 @@
+        _ => simple_response(StatusCode::NOT_FOUND, "not found"),
+    };
+
+    Ok(res)
+}
+
+/// `/healthy` endpoint handler.
+///
+/// Returns `200 OK` when the number of recently live peers is greater than or equal to
+/// `min_connected_peers`. Otherwise returns `503 Service Unavailable`.
+async fn healthy<A, Tip, SyncStatus>(ctx: &HealthCtx<A, Tip, SyncStatus>) -> Response<Full<Bytes>>
+where
+    A: AddressBookPeers + Clone + Send + Sync + 'static,
+    Tip: ChainTip + Clone + Send + Sync + 'static,
+    SyncStatus: ChainSyncStatus + Clone + Send + Sync + 'static,
+{
+    let live = ctx.address_book.recently_live_peers(Utc::now()).len();
+    if live >= ctx.config.min_connected_peers {
+        simple_response(StatusCode::OK, "ok")
+    } else {
+        simple_response(StatusCode::SERVICE_UNAVAILABLE, "no peers")
+    }
+}
+
+/// `/ready` endpoint handler.
+///
+/// Returns `200 OK` when the node is near the tip and the estimated lag is less than or equal
+/// to `ready_max_blocks_behind`. Negative lags are treated as 0. When `enforce_on_test_networks`
+/// is `false`, returns `200` on regtest/testnets regardless of state.
+async fn ready<A, Tip, SyncStatus>(ctx: &HealthCtx<A, Tip, SyncStatus>) -> Response<Full<Bytes>>
+where
+    A: AddressBookPeers + Clone + Send + Sync + 'static,
+    Tip: ChainTip + Clone + Send + Sync + 'static,
+    SyncStatus: ChainSyncStatus + Clone + Send + Sync + 'static,
+{
+    // Allow everything on test networks unless enforced
+    if !ctx.config.enforce_on_test_networks && ctx.network.is_a_test_network() {
+        return simple_response(StatusCode::OK, "ok");
+    }
+
+    if !ctx.sync_status.is_close_to_tip() {
+        return simple_response(StatusCode::SERVICE_UNAVAILABLE, "syncing");
+    }
+
+    match ctx
+        .latest_chain_tip
+        .estimate_distance_to_network_chain_tip(&ctx.network)
+    {
+        Some((distance, _local_tip)) => {
+            // Treat negative distance as 0 (ahead due to clock skew)
+            let behind: i64 = std::cmp::max(0, distance);
+            if behind <= ctx.config.ready_max_blocks_behind {
+                simple_response(StatusCode::OK, "ok")
+            } else {
+                simple_response(
+                    StatusCode::SERVICE_UNAVAILABLE,
+                    &format!("lag={} blocks", behind),
+                )
+            }
+        }
+        None => simple_response(StatusCode::SERVICE_UNAVAILABLE, "no tip"),
+    }
+}
+
+/// Creates a minimal plain-text HTTP response with the provided status.
+fn simple_response(status: StatusCode, body: &str) -> Response<Full<Bytes>> {
+    let bytes = Bytes::from(body.to_string());
+    let len = bytes.len();
+    ResponseBuilder::new()
+        .status(status)
+        .header(CONTENT_TYPE, "text/plain; charset=utf-8")
+        .header(CONTENT_LENGTH, len.to_string())
+        .body(Full::new(bytes))
+        .expect("valid response")
+}
+
+#[cfg(all(test, feature = "proptest-impl"))]
+mod tests {
```

Comment:
Nitpick: It would be nice to move this to its own file (at `components/health/tests.rs`, then move the mod declaration to the top of the file and condition it on just `cfg(test)` if it doesn't need anything that's behind `feature = "proptest-impl"`)

```suggestion
#[cfg(test)]
mod tests;
```